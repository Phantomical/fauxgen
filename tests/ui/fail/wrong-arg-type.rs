
#[fauxgen::generator(arg = i32)]
fn gen() {
    let value: &str = argument!();
}

fn main() {}
