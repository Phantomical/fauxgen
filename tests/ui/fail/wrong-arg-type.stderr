error[E0308]: mismatched types
 --> tests/ui/fail/wrong-arg-type.rs:2:1
  |
2 | #[fauxgen::generator(arg = i32)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found `i32`
3 | fn gen() {
4 |     let value: &str = argument!();
  |                       ----------- in this macro invocation
  |
  = note: this error originates in the macro `argument` (in Nightly builds, run with -Z macro-backtrace for more info)
